server/channel.py

```
@@ -1,2 +1,99 @@
+import json
+
 class NoSuchChannelError(Exception):
     pass
+
+class ChannelCreationError(Exception):
+    pass
+
+class Channel(object):
+
+    @classmethod
+    def create(cls):
+        """
+        Creates a new channel.
+        """
+        pass
+
+    def __init__(self, name, broadcasterUUID):
+        self.name = name
+        self.broadcaster = broadcasterUUID
+        self.userlist = []
+        self.url = "http://baseurl.com/" + name
+        self.add_listener(name)
+        super(Channel, self).__init__()
+
+    num_listeners = property(lambda self: len(self.userlist))
+
+    def add_listener(self, listener):
+        if not (listener in self.userlist):
+            self.userlist.append(listener)
+            return True
```

For future reference a method like this should raise an error with a meaningful message if something cannot be done. Don't return booleans like this.